groovy/gradle/src/org/netbeans/modules/gradle/ActionProviderImpl.java

```diff
@@ -250,6 +251,9 @@ private static void invokeProjectAction(final Project project, final ActionMappi
                     prj.reloadProject(true, maxQualily, reloadArgs);
                 }
                 project.getLookup().lookup(AfterBuildActionHook.class).afterAction(action, outerCtx, task.result(), out1);
+                for (TaskListener l : context.lookupAll(TaskListener.class)) {
+                    l.taskFinished(t);
```

Comment:
This is the extension created for synchronous execution of the `BUILD` and `RUN` actions in the test, @lkishalmi.

Reply:
Well, I would not leak the task out of this context. As far as I see registering an AfterBuildActionHook could do the same as this task listener magic.

Reply:
Rewritten to `AfterBuildActionHook` in cee89d4 - but it still needs a change in `ActionProviderImpl`. The previous code only searched `project.getLookup()`...

Reply:
That's actually a nice enhancement. Thank you for that!